src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsOperation.cs

```diff
@@ -132,7 +134,114 @@ private IEnumerable<IEdmOperationParameter> ComputeParameters()
                 }
             }
 
-            return parameters;
+            // Handle the out-of-line optinal parameter annotation for parameters.
+            // First, use the above built parameters to create the full target name, for example: NS.TestFunction(Edm.String, Edm.String, Edm.String)
+            // Then, go through each prameters by visiting the out-of-line optional parameter annotation.
+            // If we find at least one of out-of-line optional parameter annotation, we replace it as an optional parameter.
+            // Otherwise, re-use the built parameter.
+            // Be noted: if a parameter has inline and out-of-line optional parameter, the out-of-line will win.
+            string fullName = ParameterizedTargetName(Namespace + "." + Name, parameters);
+
+            List<IEdmOperationParameter> newParameters = new List<IEdmOperationParameter>();
+            foreach (var parameter in parameters)
+            {
+                // for example: NS.TestFunction(Edm.String, Edm.String, Edm.String)/OptionalParameter
+                string fullTargetName = fullName + "/" + parameter.Name;
+                string defaultValue;
+
+                if (TryGetOptionalParameterAnnotation(fullTargetName, out defaultValue))
+                {
+                    CsdlSemanticsOperationParameter csdlSemanticsParameter = (CsdlSemanticsOperationParameter)parameter;
+                    newParameters.Add(new CsdlSemanticsOptionalParameter(this, (CsdlOperationParameter)csdlSemanticsParameter.Element, defaultValue));
+                }
+                else
+                {
+                    newParameters.Add(parameter);
+                }
+            }
+
+            return newParameters;
+        }
+
+        private bool TryGetOptionalParameterAnnotation(string fullTargetName, out string defaultValue)
+        {
+            defaultValue = null;
+            bool isOptional = false;
+
+            List<CsdlSemanticsAnnotations> annotations;
+            if (Model.OutOfLineAnnotations.TryGetValue(fullTargetName, out annotations))

```

Comment:
>fullTargetName [](start = 55, length = 14)

Note that OData 4.0 applies annotations to all overloads of a function or action. It isn't until recently, in [ODATA-1178](https://issues.oasis-open.org/browse/ODATA-1178?jql=project%20%3D%20ODATA%20AND%20text%20~%20%22overload%22), that we added support for externally annotating different overloads, but we still support annotating the non-overloaded version (which, for this case, I think would be the common scenario).  SO, we should probably first check the fullTargetName and, if that doesn't match, check just the function name.